### ISSUE TITLE
Relative order for Consistent Help (3.2.6) when serial order is relative but not visual order

### ISSUE THREAD

**Issue by mraccess77** [w3c/wcag#2299]
The SC implies that within a page variation the visual order should be relative, but it's not required if the serial order is correct.  The note however is aimed at variations.

The note in the understanding doc states "For this SC, the same relative order can be thought of as how the content is ordered when the page is serialized. The visual position of a help mechanism is likely to be consistent across pages for the same page variation (e.g., CSS break-point). The user can cause a change, for example zoom or change orientation, which may trigger a different page variation."

I've seen instances where in the same responsive variation the help mechanism is visually in different locations but serially in the same order.  That is on one page it's at the end of content but before the footer in the main area and on another page it's also between the content and the footer but on the right side (same variation/viewport).  Does that meet the SC as the note implies?  It seems like it would be best otherwise - but would pass.

**Comment by alastc** [comment]
> in the same responsive variation the help mechanism is visually in different locations but serially in the same order

If it is between the main content and footer across pages, but sometimes on the left sometimes on the right (for example), I think that would pass. Not good from a user point of view, but I don't think we can narrow it down to catch those scenarios without creating different issues elsewhere.

**Comment by mraccess77** [comment]
Thanks Alastair, I'm fine with that - but I want to make sure others in the group agree on this because I don't want to pass something 80% of the group is failing.

**Comment by patrickhlauke** [comment]
coming in late, but agree that it's the serial order that matters, not the visual order (though perhaps that should be emphasised further, with an example)

**Comment by alastc** [comment]
Well, I [emailed the group](https://lists.w3.org/Archives/Public/w3c-wai-gl/2022AprJun/0189.html), and no-one disagreed. Are you happy to close this, or should we keep it open for an understanding doc update?

**Comment by mraccess77** [comment]
I would prefer it be documented in the understanding document - but we do have this issue as proof of the discussion if we need to close it.

**Comment by bruce-usab** [comment]
> That is on one page it's at the end of content but before the footer in the main area and on another page it's also between the content and the footer but on the right side (same variation/viewport).

Sorry, but this examples seems incompatible to be with (emphasis added) <q>they occur in the same relative order **to other page content**</q>.

In my mind, I was thinking the updated phrasing for 3.2.6 was essentially editorial.  With JonA's example, we might be precluding more than we had expected.

**Comment by alastc** [comment]
Hi Bruce,

We've been through the difficulty of trying to specify where something is in a layout, that's why we ended up with "the same relative order", like consistent navigation.

So it's a proxy for being in a same or similar place visually, but it's not 1:1. That's the logic of the SC text.

"The visual position of a help mechanism is **_likely_** to be consistent", but it won't be 100%.

It's not perfect, but I don't see another option.

**Comment by bruce-usab** [comment]
@alastc -- Apologies, I agree its fine.  I somehow read:

> end of content but before the footer in the main area

as a different serialization than:

> and on another page it's also between the content and the footer

I misunderstood Jon's example to be saying the different visual presentation resulted in an insertion between [content] [help] [footer].

**Comment by alastc** [comment]
@bruce-usab volunteered for a small understanding doc update.

**Comment by alastc** [comment]
I've added line 53 to PR #2998 to address this.
